### ISSUE TITLE
Branch is incorrect for git:// URLs

### ISSUE THREAD

**Issue by mattdesl** [zeke/github-url-to-object#20]
`git://github.com/mattdesl/img#add-options`

The above gets parsed correctly by npm / package.json dependency resolution. However, this module returns the following, with `master` branch:

```js
{ user: 'mattdesl',
  repo: 'img',
  host: 'github.com',
  branch: 'master',
  apiHost: 'api.github.com',
  tarball_url: 'https://api.github.com/repos/mattdesl/img/tarball/master',
  clone_url: 'https://github.com/mattdesl/img',
  https_url: 'https://github.com/mattdesl/img',
  travis_url: 'https://travis-ci.org/mattdesl/img',
  zip_url: 'https://github.com/mattdesl/img/archive/master.zip',
  api_url: 'https://api.github.com/repos/mattdesl/img' }
```

**Comment by zeke** [comment]
I don't think this is an issue with `git://` URLs. The shorthand `#commitish` fragment syntax may be supported by the npm client and other tools, but it's not officially a part of github URLs.

The shorthand commitish syntax is supported if you're going "fully shorthand", as in `  gh('github:monkey/business#nachos')`, but not recognized as a branch when you pass it a fully-qualified URL.

I'd happily accept a PR that adds support for this syntax.
